tests/test_core.py

```diff
@@ -9341,8 +9341,19 @@ def test_abort_on_exceptions(self):
     self.set_setting('ALLOW_TABLE_GROWTH')
     self.set_setting('EXPORTED_RUNTIME_METHODS', ['ccall', 'cwrap'])
     self.set_setting('DEFAULT_LIBRARY_FUNCS_TO_INCLUDE', ['$addFunction'])
-    self.emcc_args += ['-lembind', '--post-js', test_file('core/test_abort_on_exception_post.js')]
-    self.do_core_test('test_abort_on_exception.cpp', interleaved_output=False)
+    self.emcc_args += ['-lembind', '--post-js', test_file('core/test_abort_on_exceptions_post.js')]
+    self.do_core_test('test_abort_on_exceptions.cpp', interleaved_output=False)
+
+  def test_abort_on_exceptions_main(self):
+    # The unhandled exception wrappers should not kick in for exceptions thrown during main
```

Comment:
Why is that the behavior? `ABORT_ON_WASM_EXCEPTIONS` docs don't say anything special about `main()`, and it seems like `main()` should be handled the same?

Reply:
If you read the comment in preamble.js around `ABORT_ON_WASM_EXCEPTIONS` and you look at the current beahviour of callMain its clear that the intent is not to interfere with the try/catch behaviour that already exists in main.  i.e. main already handles exceptions and aborts/exits appropriately.  `ABORT_ON_WASM_EXCEPTIONS` kicks in when exceptions a happen on otherwise unprotected entry points into the module.  

I'm not changing the behaviour here, I'm just adding a missing test.

Reply:
Oh, are the JS changes NFC?

If this is the old behavior then please clarify that in settings.js (that is what confused me).

Reply:
Yes, they are NFC, its a little but of a cleanup, and I noticed the code wasn't tested so I added a test.

Reply:
I can land the test first if you prefer ?

Reply:
No need to split it out. lgtm with a clarification in settings.js (that main is handled differently).

Reply:
Updated docs.